utils/hare-fetch-fids

```diff
@@ -106,16 +112,19 @@ def get_service_for_node(nodes: List[Node], node_name: str = None, service: str
     if service == None:
         for node in nodes:
             if node.name == node_name:
-                return simplejson.dumps(node, indent=2, for_json=True)
+                return node.svcs
 
+    # returns all the services for the specified given service
     for node in nodes:
         if node.name == node_name:
             return node.get_service(service)
 
-    print(f'Node {node_name} not found in cluster.', file=sys.stderr)
+    print(
+        f'Node {node_name} not found in cluster.', file=sys.stderr)
+    return None
```

Comment:
[Optional]: usually it is not the best practice to return None; this forces the caller to make additional check (if not None is returned) - more code to write and one more point where a mistake can happen.

In other words, returned value here aims two different things: return the result of some computation AND report about an error. But why do we need the latter if the language supports exceptions? You can throw an exception if something unexpected is happening. Otherwirse return the actual value.

But I don't require it for this script, I think we can merge it as it is now.

Reply:
Oh yes, thanks a lot!

> You can throw an exception if something unexpected is happening. Otherwirse return the actual value.
Makes sense. Will keep in mind.